packages/manager/src/managers/sliceTemplateLibrary/SliceTemplateLibraryManager.ts

```diff
@@ -0,0 +1,179 @@
+import * as t from "io-ts";
+import { HookError } from "@slicemachine/plugin-kit";
+import { BaseManager } from "../BaseManager";
+import { DecodeError } from "../../lib/DecodeError";
+import { assertPluginsInitialized } from "../../lib/assertPluginsInitialized";
+import { decodeHookResult } from "../../lib/decodeHookResult";
+import { SharedSlice } from "@prismicio/types-internal/lib/customtypes";
+import { SharedSliceContent } from "@prismicio/types-internal/lib/content";
+
+type SliceTemplateLibraryManagerReadReturnType = {
+	templates: {
+		model: SharedSlice;
+		screenshots: Record<string, Buffer>;
+	}[];
+	errors: (DecodeError | HookError)[];
+};
+
+type SliceTemplateLibraryManagerCreateReturnType = {
+	data?: {
+		sliceIds: string[];
+	};
+	errors: (DecodeError | HookError)[];
+};
+
+type SliceTemplateLibraryReadData = {
+	templateIds?: string[];
+};
+
+type SliceTemplateLibraryCreateData = SliceTemplateLibraryReadData;
+
+const readHookCodec = t.type({
+	templates: t.array(
+		t.type({
+			model: SharedSlice,
+			componentContents: t.string,
+			mocks: t.array(SharedSliceContent),
+			screenshots: t.record(t.string, t.any),
+		}),
+	),
+});
+
+export class SliceTemplateLibraryManager extends BaseManager {
+	async read(
+		args: SliceTemplateLibraryReadData,
+	): Promise<SliceTemplateLibraryManagerReadReturnType> {
+		assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+		const isTypeScriptProject = await this.project.checkIsTypeScript();
+		const hookResult = await this.sliceMachinePluginRunner.callHook(
+			"slice-template-library:read",
+			{ ...args, isTypeScriptProject },
+		);
+
+		const { data, errors } = decodeHookResult(readHookCodec, hookResult);
+
+		return {
+			templates: data.flat()[0].templates,
+			errors,
+		};
+	}
+	async create(
```

Comment:
The "create" name implies that it is creating a slice template library, but that is not the case here.

Should this logic be part of SliceManager instead?

Reply:
Personally I feel like keeping all things relative to slices templates in SliceTemplateLibrary manager, easy to maintain, but maybe that's not the idea?

Reply:
Works for me, especially with the method rename. I just wanted to make sure we considered it.